assets/scripts/app/ScrollIndicators.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { registerKeypress } from './keypress'
+
+class ScrollIndicators extends React.Component {
+  static propTypes = {
+    posLeft: PropTypes.number.isRequired,
+    posRight: PropTypes.number.isRequired,
+    scrollStreet: PropTypes.func.isRequired,
+    scrollTop: PropTypes.number.isRequired
+  }
+
+  shouldComponentUpdate (nextProps) {
+    if (this.props.posLeft !== nextProps.posLeft &&
+        this.props.posRight !== nextProps.posRight) {
+      this.refs.left_indicator.innerHTML = Array(nextProps.posLeft + 1).join('‹')
+      this.refs.right_indicator.innerHTML = Array(nextProps.posRight + 1).join('›')
+      return true
+    }
+    // Checking if window was resized
+    if (this.props.scrollTop !== nextProps.scrollTop) {
+      this.updateIndicatorsPosition(nextProps.scrollTop)
+      return true
+    }
+    return false
+  }
+
+  componentDidMount () {
+    registerKeypress('left', (event) => { this.props.scrollStreet(true, event.shiftKey) })
+    registerKeypress('right', (event) => { this.props.scrollStreet(false, event.shiftKey) })
+  }
+
+  updateIndicatorsPosition = (scrollTop) => {
+    this.refs.left_indicator.style.top = scrollTop + 'px'
+    this.refs.right_indicator.style.top = scrollTop + 'px'
+  }
+
+  render () {
+    return (
+      <div className="street-scroll-indicators">
+        <div id="street-scroll-indicator-left" onClick={(e) => this.props.scrollStreet(true, e.shiftKey)} ref="left_indicator" />
+        <div id="street-scroll-indicator-right" onClick={(e) => this.props.scrollStreet(false, e.shiftKey)} ref="right_indicator" />
```

Comment:
Ideally we don't need the refs anymore (see above) and we also don't need the `id`s anymore, but for styling purposes we should convert those to `className`s.

Reply:
hey @louh, I tried to change the `id`s to `className`s but it ends up breaking the loading page of Streetmix. It just infinitely remains on the 3/4 mark of the loading bar. 
I tried finding other places in the code that is looking for `#street-scroll-indicator-left` or `#street-scroll-indicator-right` that might be causing this but I can't find any... 
Any ideas why it's affecting the loading bar?

Reply:
Breaking the loading page?! Wow. OK, so assuming you tried clearing the cache/reloading and it's still causing the problem, then something is probably more tightly coupled to this than I expected. (Is there an error message?) Let's leave this alone for now, but file an issue for us to look at later.

Reply:
Yea, I reloaded/cleared the cache and it still breaks the loading page. No error message either. 
Okay I will file an issue.